Services/AspireService.cs

```diff
@@ -1,93 +1,146 @@
-                await responseStream.WriteAsync(logsUpdate);
-                // Introduce delay for testing
-                await Task.Delay(1000);
-            }
-        }
-
-        public override Task<ResourceCommandResponse> ExecuteResourceCommand(ResourceCommandRequest request, ServerCallContext context)
-        {
-            return base.ExecuteResourceCommand(request, context);
-        }
-
-    }
-}
+            Console.WriteLine($"WatchResources connection canceled");
+        }
+
+        private async Task WriteInitialDataToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Gets the initial data and return it
+
+            var podsList = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace);
+            foreach (var pod in podsList)
+            {
+                var labels = pod.Labels();
+                if (labels.IsNullOrEmpty())
+                {
+                    _logger.LogDebug($"Unable to fetch app name because Pod: {pod.Metadata.Name} does not contain labels.Skipping to next pod.");
+                    continue;
+                }
+
+                var result = labels.TryGetValue("containerapps.io/app-name", out string appName);
+                if (!result)
+                {
+                    _logger.LogDebug($"The Pod {pod.Metadata.Name} does not have an entry for the key app. Skipping to next pod.");
+                    continue;
+                }
+
+                _logger.LogDebug($"App name: {appName}");
+                _logger.LogDebug($"Result: {result}");
+                var initialData = new WatchResourcesUpdate()
+                {
+                    InitialData = new InitialResourceData()
+                    {
+                        Resources = { new Resource() { Name = appName, Commands = { new ResourceCommandRequest() { CommandType = "Restart" } } } }
+                    }
+                };
+
+                await responseStream.WriteAsync(initialData);
+            }
+        }
+
+        private async Task WatchAndWriteChangesToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Creates the watcher
+            var podsWatchResponse = await _kubernetesClient.CoreV1.ListNamespacedPodWithHttpMessagesAsync(_kubernetesConfig.Namespace, watch: true);
+            var podWatcher = podsWatchResponse.Watch<V1Pod, V1PodList>(async (type, item) =>
+            {
+                Console.WriteLine($"Pod event of type {type} detected for {item.Metadata.Name}");
+                if (type.Equals(WatchEventType.Added) || type.Equals(WatchEventType.Modified))
+                {
+                    var watchResourcesUpdate = new WatchResourcesUpdate()
+                    {
+                        Changes = new WatchResourcesChanges()
+                        {
+                            Value = { new WatchResourcesChange() { Upsert = new Resource { Name = item.Metadata.Name } } }

```

Comment:
You might already be working on this so please ignore: could you please add the pod data there as well?

Reply:
Added pod data in my latest commit.